app/services/trust_update_service.rb

```diff
@@ -4,18 +4,48 @@ def update_trusts(last_update: nil)
     last_update ||= DataStage::DataUpdateRecord.last_update_for(:trusts)
 
     # simple updates for trusts that are open
+    update_open_trusts(last_update)
+
+    # auto close trusts that have no schools
+    # skip and notify sentry when a trust is closed and still has schools - needs investigation
+    close_trusts(last_update)
+
+    DataStage::DataUpdateRecord.updated!(:trusts)
+  end
+
+private
+
+  def update_open_trusts(last_update)
     DataStage::Trust.updated_since(last_update).gias_status_open.each do |staged_trust|
       trust = Trust.find_by(companies_house_number: staged_trust.companies_house_number)
 
       next unless trust
 
       update_trust(trust, staged_trust)
     end
-
-    DataStage::DataUpdateRecord.updated!(:trusts)
   end
 
-private
+  def close_trusts(last_update)
+    trusts_with_schools = []
+
+    DataStage::Trust.updated_since(last_update).gias_status_closed.each do |staged_trust|
+      trust = Trust.find_by(companies_house_number: staged_trust.companies_house_number)
+
+      next unless trust
+      next if trust.status == 'closed'
+
+      if trust.schools.count.positive?
+        trusts_with_schools << trust.id
+        next
+      end
+
+      close_trust(trust)
+    end
+
+    unless trusts_with_schools.empty?
+      Sentry.capture_message("TrustUpdateService#close_trusts - FAILED TO CLOSE: #{trusts_with_schools.inspect}")
```

Comment:
Do we have an example of what this will look like in sentry? I suppose if it includes the stack we can easily deduce that it failed because the Trust has schools. Otherwise we would need to add that fact to the capture_message.

Reply:
Here is an example from dev: https://sentry.io/organizations/dfe-get-help-with-tech/issues/2411110736/

Reply:
Yep I shared a link in Slack.

It's a custom message so we know exactly where it's from, not much need for a Stacktrace as it's not an exception we are handling.